From: Marc Kleine-Budde <mkl@pengutronix.de>
Subject: remove __depreacted tag

This patch fixes this problem, when compiling glibc-2.7 on arm with sanitized kernel-header:

arm-v4t-linux-gnueabi-gcc ../ports/sysdeps/unix/sysv/linux/arm/ioperm.c -c -std=gnu99 -fgnu89-inline -O2 -Wall -Winline -Wwrite-strings -fmerge-all-constant
s -g -Wstrict-prototypes      -I../include -I/home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/build-target/glibc-2.7-first-build/misc -I/home/frogge
r/pengutronix/toolchain/OSELAS.Toolchain-mkl/build-target/glibc-2.7-first-build -I../ports/sysdeps/arm/elf -I../ports/sysdeps/unix/sysv/linux/arm/eabi/nptl
-I../ports/sysdeps/unix/sysv/linux/arm/eabi -I../ports/sysdeps/unix/sysv/linux/arm/nptl -I../ports/sysdeps/unix/sysv/linux/arm -I../ports/sysdeps/unix/sysv/
linux -I../nptl/sysdeps/unix/sysv/linux -I../nptl/sysdeps/pthread -I../sysdeps/pthread -I../sysdeps/unix/sysv/linux -I../sysdeps/gnu -I../sysdeps/unix/commo
n -I../sysdeps/unix/mman -I../sysdeps/unix/inet -I../ports/sysdeps/unix/sysv -I../nptl/sysdeps/unix/sysv -I../sysdeps/unix/sysv -I../ports/sysdeps/unix/arm
-I../ports/sysdeps/unix -I../nptl/sysdeps/unix -I../sysdeps/unix -I../sysdeps/posix -I../ports/sysdeps/arm/eabi -I../ports/sysdeps/arm/nptl -I../ports/sysde
ps/arm -I../sysdeps/wordsize-32 -I../sysdeps/ieee754/flt-32 -I../sysdeps/ieee754/dbl-64 -I../sysdeps/ieee754 -I../sysdeps/generic/elf -I../sysdeps/generic -
I../ports -I../nptl  -I.. -I../libio -I. -nostdinc -isystem /home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/opt/OSELAS.Toolchain-trunk/arm-v4t-linu
x-gnueabi/gcc-4.2.3-glibc-2.7-kernel-2.6.25-sanitized/gcc-first/lib/gcc/arm-v4t-linux-gnueabi/4.2.3/include -isystem /home/frogger/pengutronix/toolchain/OSE
LAS.Toolchain-mkl/opt/OSELAS.Toolchain-trunk/arm-v4t-linux-gnueabi/gcc-4.2.3-glibc-2.7-kernel-2.6.25-sanitized/sysroot-arm-v4t-linux-gnueabi/usr/include -D_
LIBC_REENTRANT -include ../include/libc-symbols.h       -o /home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/build-target/glibc-2.7-first-build/misc/
ioperm.o -MD -MP -MF /home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/build-target/glibc-2.7-first-build/misc/ioperm.o.dt -MT /home/frogger/pengutro
nix/toolchain/OSELAS.Toolchain-mkl/build-target/glibc-2.7-first-build/misc/ioperm.o
In file included from /home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/opt/OSELAS.Toolchain-trunk/arm-v4t-linux-gnueabi/gcc-4.2.3-glibc-2.7-kernel-2
.6.25-sanitized/sysroot-arm-v4t-linux-gnueabi/usr/include/asm/page.h:178,
                 from ../ports/sysdeps/unix/sysv/linux/arm/ioperm.c:48:
/home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/opt/OSELAS.Toolchain-trunk/arm-v4t-linux-gnueabi/gcc-4.2.3-glibc-2.7-kernel-2.6.25-sanitized/sysroo
t-arm-v4t-linux-gnueabi/usr/include/asm/memory.h:191: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'unsigned'
/home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/opt/OSELAS.Toolchain-trunk/arm-v4t-linux-gnueabi/gcc-4.2.3-glibc-2.7-kernel-2.6.25-sanitized/sysroo
t-arm-v4t-linux-gnueabi/usr/include/asm/memory.h:196: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'void'
make[3]: *** [/home/frogger/pengutronix/toolchain/OSELAS.Toolchain-mkl/build-target/glibc-2.7-first-build/misc/ioperm.o] Error 1

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>

---
 include/asm-arm/memory.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.25/include/asm-arm/memory.h
===================================================================
--- linux-2.6.25.orig/include/asm-arm/memory.h
+++ linux-2.6.25/include/asm-arm/memory.h
@@ -188,12 +188,12 @@ static inline void *phys_to_virt(unsigne
  * memory.  Use of these is *deprecated* (and that doesn't mean
  * use the __ prefixed forms instead.)  See dma-mapping.h.
  */
-static inline __deprecated unsigned long virt_to_bus(void *x)
+static inline unsigned long virt_to_bus(void *x)
 {
 	return __virt_to_bus((unsigned long)x);
 }
 
-static inline __deprecated void *bus_to_virt(unsigned long x)
+static inline void *bus_to_virt(unsigned long x)
 {
 	return (void *)__bus_to_virt(x);
 }
